client/my-sites/plugins/plugin-meta/index.jsx

```diff
@@ -404,16 +405,19 @@ export class PluginMeta extends Component {
 	}
 
 	getAvailableNewVersions() {
+		const { pluginsOnSites } = this.props;
 		return this.props.sites
 			.map( ( site ) => {
 				if ( ! site.canUpdateFiles ) {
 					return null;
 				}
-				if ( site.plugin && site.plugin.update ) {
-					if ( 'error' !== site.plugin.update && site.plugin.update.new_version ) {
+
+				const sitePlugin = pluginsOnSites?.sites[ site.ID ];
+				if ( sitePlugin?.update ) {
+					if ( 'error' !== sitePlugin.update?.new_version ) {
```

Comment:
nit: does it make sense to simplify this to only 1 if statement?

Reply:
Of course 👍 Nice find. Fixed in https://github.com/Automattic/wp-calypso/pull/48482/commits/80c27015b92fb511e3a7c7df666c71eba89c1278